### PR TITLE
Eval framework

### PR DESCRIPTION
**Pull request by KaylaKwon**
@ethkim @nlpenguin @ifndefdeadmau5 @djflexible 

Eval Framework completed. We'll update survey data from now on.
Unit test is not yet developed. (will make soon!)

### CODE REVIEW

File: src/server/config/json/survey.data.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Sample Data for survey test
```
Comment (ethkim) [inline src/server/config/json/survey.data.js]: If this gets used in the test, it should live in test directory.

File: src/server/config/mongoose.js [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+import '../models/survey.model';
 import '../models/users.model';
 import '../models/match.model';
 
 export default function () {
   mongoose.Promise = global.Promise;
-  const db = mongoose.connect('mongodb://localhost:27017/yoda');
+  const db = {
+    development: mongoose.connect('mongodb://localhost:27017/yoda'),
+    test: 'mongodb://localhost:27017/yoda-test',
```
Comment (ethkim) [inline src/server/config/mongoose.js]: Is this a new change that was made in another PR?

File: src/server/controllers/survey.controller.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ * Methods about Eval Framework, save or return survey/answer
+ */
+
+const Answer = mongoose.model('answer');
+const Survey = mongoose.model('survey');
+const User = mongoose.model('user');
+
+// Get request
+export function getRequest(req, res, next) {
+  let t = Math.random() * 100; // random number between 0 ~ 101
```
Comment (ethkim) [inline src/server/controllers/survey.controller.js]: Add todo to refactor out this traffic teeing method.

File: src/server/controllers/survey.controller.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import mongoose from 'mongoose';
```
Comment (ethkim) [inline src/server/controllers/survey.controller.js]: There are no unit tests for any of these methods. Are they coming in a separate PR?

File: src/server/controllers/users.controller.js [hunk 5]
```diff
@@ -33,9 +37,9 @@ export function getMentorList(req, res, next) {
       // TODO: Longer term, we should migrate to a UserSummary object
       // that contains subset of fields. For now, we return all fields.
       if (err) {
-        res.send(err);
+        res.status(400).send(err);
```
Comment (ethkim) [inline src/server/controllers/users.controller.js]: Why was this included in this PR? Was this from another PR?

File: src/test/controllers/survey.controller.test.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+// import '../../server/models/answer.model';
```
Comment (ethkim) [inline src/test/controllers/survey.controller.test.js]: Don't include this in this PR if you're going to leave them commented out. Or just complete the tests :)

File: src/server/config/mongoose.js (hunk 2 above)
Comment (SarahhhC) [inline src/server/config/mongoose.js]: nope, I think another feature added. This is for test db setting...

File: src/server/controllers/survey.controller.js (hunk 4 above)
Comment (SarahhhC) [inline src/server/controllers/survey.controller.js]: Unit test is not yet completed, so it will be next PR...

File: src/server/controllers/users.controller.js (hunk 5 above)
Comment (SarahhhC) [inline src/server/controllers/users.controller.js]: We decided to handle err as same way, so just sync with other back-end.

**Comment by ethkim** [comment]
Without Unit tests or frontend part complete, it's hard to know if this work is done. Please update the tests while the frontend is under way. 
Thanks guys! LGTM.
